### PR TITLE
STORM-2748: Fix TickTupleTest to actually test something

### PR DESCRIPTION
**Pull request by revans2**

### CODE REVIEW

File: storm-server/src/test/java/org/apache/storm/TickTupleTest.java [hunk 1]
```diff
@@ -18,85 +18,103 @@
...
+        }
     }
 
-    private BaseRichBolt makeNoOpBolt() {
-        return new BaseRichBolt() {
-            @Override
-            public void prepare(Map<String, Object> conf, TopologyContext topologyContext, OutputCollector outputCollector) {}
-            @Override
-            public void execute(Tuple tuple) {}
+    private static class NoopBlot extends BaseRichBolt {
```
Comment (srdo) [inline storm-server/src/test/java/org/apache/storm/TickTupleTest.java]: Nit: Blot -> Bolt

File: storm-server/src/test/java/org/apache/storm/TickTupleTest.java [hunk 2]
```diff
@@ -18,85 +18,103 @@
...
+        @Override
+        public void prepare(Map<String, Object> conf, TopologyContext topologyContext, OutputCollector outputCollector) {}
 
-            @Override
-            public void cleanup() { }
+        @Override
+        public void execute(Tuple tuple) {
+            LOG.error("GOT {}", tuple);
```
Comment (srdo) [inline storm-server/src/test/java/org/apache/storm/TickTupleTest.java]: Nit: Info or debug might be nicer so a grep for errors in the logs doesn't produce this line.

**Comment by revans2** [comment]
@srdo thanks for the review.  Trying to get the test to work on the slower VMs is still a work in progress.  I'll let you know why I actually have it passing consistently.

**Comment by revans2** [comment]
@srdo I finally have the test passing predictably on travis.  I also fixed the issues that you saw.  Feel free to take another look.

File: storm-server/src/test/java/org/apache/storm/TickTupleTest.java [hunk 3]
```diff
@@ -18,85 +18,113 @@
...
+                //Give the cluster some time to come up
+                long time = 0;
+                while (tickTupleTimes.size() <= 0) {
+                    assert time <= 100_000 : "took over " + time + " ms of simulated time to get a message back...";
+                    cluster.advanceClusterTime(10);
+                    time += 10_000;
+                }
+                tickTupleTimes.clear();
+                cluster.advanceClusterTime(1);
+                time += 1000;
```
Comment (srdo) [inline storm-server/src/test/java/org/apache/storm/TickTupleTest.java]: I think this can be put in a loop now, the following lines don't differ anymore.

**Comment by revans2** [comment]
@srdo I turned it into a loop like you requested

**Comment by srdo** [comment]
Thanks

**Comment by HeartSaVioR** [comment]
+1

**Pull request merged**